### ISSUE TITLE
FEAT - Add GroupLasso and GroupLogisticRegression estimators 

### ISSUE THREAD

**Issue by mathurinm** [scikit-learn-contrib/skglm#197]

**Comment by tomaszkacprzak** [comment]
Hello, following up on the [issue](https://github.com/mathurinm/celer/issues/290) from Celer, it would be great to be able to run GroupLasso with non-neg weights. The standard Lasso already uses `positive=True`. Would that be possible to implement. 

For my application I also need to run on sparse X input matrices. From what I have seen, some of the regressors can handle the `csc` matrices. Would that be possible for GroupLasso?

**Comment by QB3** [comment]
Hello,
Thanks a lot for your interest!
- adding a group Lasso with positive weights should be "easy", we should be able to do it quickly (i.e. this week if we have time)
- handling sparse matrices for the group Lasso is definitely doable, but might take a bit more time to check that the implementation is correct

**Comment by tomaszkacprzak** [comment]
Thank you, I am glad that it is doable. Is there anything I can do to help?

**Comment by mathurinm** [comment]
@tomaszkacprzak we can start by supporting sparse matrices for the GroupBCD solver. Do you think you could send a PR doing that ? 

You'd need to implement `construct_grad_sparse` and `bcd_epoch_sparse` in `solvers/group_bcd.py`. They should do the same thing as their existing dense versions (same name without `_sparse`), but when X is a csc matrix, passed as a triplet X_data, X_indices, X_indptr

See current implementations for CD (not BCD) in `solvers/anderson_cd.py`: sparse version https://github.com/scikit-learn-contrib/skglm/blob/main/skglm/solvers/anderson_cd.py#L313 vs dense version https://github.com/scikit-learn-contrib/skglm/blob/main/skglm/solvers/anderson_cd.py#L272

Then inside the solver we just have an if when doing the epoch of coordinate descent : https://github.com/scikit-learn-contrib/skglm/blob/main/skglm/solvers/anderson_cd.py#L147

Do not hesitate to start a very WIP PR, so that we can discuss over there!

**Comment by QB3** [comment]
In the meantime, I will open a separate PR for the positive weights

**Comment by QB3** [comment]
Do we want a `GroupLasso` estimator [as well](https://github.com/mathurinm/celer/blob/c1b564c3c0c8fc214deb67b57f4a4dd95f66121f/celer/dropin_sklearn.py#L994) ?

**Comment by mathurinm** [comment]
@tomaszkacprzak did you try our implementation of `positive` for the Group Lasso ? ANd would you be interested in implementing sparse X support for group ?

**Comment by tomaszkacprzak** [comment]
@mathurinm I noticed that in `test_group.py` [L79](https://github.com/Badr-MOUFAD/skglm/blob/e5f0984df394cec32bf1e36061b16146476265c9/skglm/tests/test_group.py#L79) the random seed is set so that for `positive=True` all come out 0. When I change the random seed, some positive weights appear, and the error is thrown.

Example for random seed 12323

```
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 5 / 50 (10%)
Max absolute difference: 0.46942555
Max relative difference: 0.22505575
 x: array([0.      , 0.      , 0.      , 0.      , 0.      , 0.183918,
       0.      , 0.      , 0.      , 0.      , 0.      , 0.      ,
       0.      , 0.      , 0.      , 0.295971, 0.      , 0.      ,...
 y: array([0.      , 0.      , 0.      , 0.      , 0.      , 0.      ,
       0.      , 0.      , 0.      , 0.      , 0.      , 0.      ,
       0.      , 0.      , 0.      , 0.241598, 0.      , 0.      ,...
```

Any suggestions what can cause this?

**Comment by QB3** [comment]
Thanks for the report, this seems to show that the `positive=True` feature introduced in https://github.com/scikit-learn-contrib/skglm/commit/1dcbfffa91999c68e23ab76c8b5dc00456766705 was not correct.
I will investigate this week.

**Comment by Badr-MOUFAD** [comment]
Yes indeed @QB3, I open a PR #225

**Comment by tomaszkacprzak** [comment]
Great, I just checked it out, works well. Thanks @Badr-MOUFAD.  I'll start a PR for sparse X.

**Comment by mathurinm** [comment]
Fixed by #228
